packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/dynamic/DynamicMutableRealmObjectImpl.kt

```diff
@@ -20,17 +20,79 @@ import io.realm.kotlin.dynamic.DynamicMutableRealmObject
 import io.realm.kotlin.internal.RealmObjectHelper
 import io.realm.kotlin.types.RealmList
 import io.realm.kotlin.types.RealmSet
+import kotlin.reflect.KClass
 
 internal class DynamicMutableRealmObjectImpl : DynamicMutableRealmObject, DynamicRealmObjectImpl() {
 
+    override fun <T : Any> getValue(propertyName: String, clazz: KClass<T>): T {
+        // dynamicGetSingle checks nullability of property, so null pointer check raises appropriate NPE
+        return RealmObjectHelper.dynamicGet(
+            this.`io_realm_kotlin_objectReference`!!,
+            propertyName,
+            clazz,
+            nullable = false,
+            issueDynamicMutableObject = true
```

Comment:
Wouldn't these somehow already be given by `clazz`?

Reply:
Nope, `clazz` is the property type, e.g. `String` or `RealmAny`.